### PR TITLE
Include setup_requires in package setup

### PR DESCRIPTION
**Pull request by larrybradley**
This PR adds `numpy` and `Cython` to `setup_requires` for those building the package from source.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ env:
         - SETUP_CMD='test'
         - EVENT_TYPE='push pull_request'
         - SETUP_XVFB=True
+        - MPLBACKEND=Agg
```
[the remark this answers is not in the document]
Comment (larrybradley) [reply .travis.yml]: Ah, OK.  I was experimenting to see if this fixes it.  I'll remove it.

[the remark this answers is not in the document]
Comment (bsipocz) [reply .travis.yml]: it fixed it here: https://travis-ci.org/astropy/photutils/builds/463423702

[the remark this answers is not in the document]
Comment (larrybradley) [reply .travis.yml]: Thanks!

[the remark this answers is not in the document]
Comment (larrybradley) [reply .travis.yml]: @bsipocz It appears that the `MPLBACKEND` is not being set in ci-helpers.  The tests (including your `astropy-helpers` 3.1 update) are failing again.

[the remark this answers is not in the document]
Comment (bsipocz) [reply .travis.yml]: yes, it seems that the fix wasn't appropriate. Can I have this PR as a testbed for today?

[the remark this answers is not in the document]
Comment (larrybradley) [reply .travis.yml]: Yes, thanks.  Is there any harm in explicitly defining `MPLBACKEND` in the `photutils` travis config?

[the remark this answers is not in the document]
Comment (bsipocz) [reply .travis.yml]: no harm, but it should have worked already from the ci-helpers

[the remark this answers is not in the document]
Comment (larrybradley) [reply .travis.yml]: @bsipocz I'm trying to get a new release out (and other PRs are forthcoming), so I'm going to go ahead and explicitly define `MPLBACKEND` here.

[the remark this answers is not in the document]
Comment (larrybradley) [reply .travis.yml]: Are you going to look at this today?  I'm hoping to release 0.6 today, and this PR should be included.

**Pull request merged**